packages/nest/src/exclusive-field.validator.ts

```diff
@@ -0,0 +1,102 @@
+import { registerDecorator, ValidationArguments } from 'class-validator';
+
+/**
+ * Decorator to ensure that only one of the fields in a class exists, or one
+ * of the fields in a select group of fields in the class exists.
+ *
+ * @param fields - An optional array of fields or an array of an array of fields
+ * of which each grouping is mutually exclusive. If any of the fields in
+ * each group are present, the others must be null or undefined.
+ *
+ * Given the following class:
+ *
+ * class CharacterFealty {
+ *     orcs?: boolean;
+ *     elves?: boolean;
+ *     humans?: boolean;
+ *     dwarves?: boolean;
+ *   }
+ *
+ * @example
+ * @ExclusiveFieldExists()
+ * fealty: CharacterFealty;
+ *
+ * The existence of any of the fields in CharacterFealty is mutually exclusive.
+ *
+ * @example
+ * @ExclusiveFieldExists(['orcs', 'elves'])
+ * fealty: CharacterFealty;
+ *
+ * The existence of 'orcs' and 'elves is mutually exclusive
+ *
+ * @example
+ * @ExclusiveFieldExists([['orcs', 'elves'],['humans', 'dwarves']])
+ * fealty: CharacterFealty;
+ *
+ * The existence of 'orcs' and 'elves is mutually exclusive, as is the existence
+ * of 'humans' and 'dwarves'. However, 'orcs' and 'humans' can exist together.
+ */
+export const ExclusiveFieldExists =
+  (fields?: string[] | string[][]): PropertyDecorator =>
+  (object: any, propertyName: string | symbol) => {
+    registerDecorator({
+      name: 'exclusiveFieldExists',
+      target: object.constructor,
+      propertyName: propertyName as string,
+      constraints: fields as any[],
+      validator: {
+        validate: (val, args: ValidationArguments) => {
+          const { value } = args;
+          let { constraints } = args;
+          const suppliedKeys = Object.entries(value).flatMap(([key, val]) =>
+            val != null ? key : [],
+          );
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          if (constraints) {
+            constraints = constraints.some((item) => Array.isArray(item))
```

Comment:
This exists solely to allow the convenience of passing a single array if there is only one grouping, instead of having to do something like this:  `[['wizard', 'troll']]`.  If there's a better way, I'm all 👂🏼 s.

Reply:
You could use a variadic arg.
```ts
(...groups: Array<readonly string[]>): PropertyDecorator
```
```ts
@ExclusiveFieldExists(['orcs', 'elves'])
@ExclusiveFieldExists(['orcs', 'elves'], ['humans', 'dwarves'])
```

That does prevent us from adding another optional arg later, though. So you could `Many<readonly string[]>` (same as what you have already just better typed).
Then you'll still need to convert. Here's how I just did that with `sortBy`
https://github.com/SeedCompany/libs/blob/f6383ad3d2601758aa01b05bd4b7ae72dd17c100/packages/common/src/sort-by.ts#L20-L20
https://github.com/SeedCompany/libs/blob/f6383ad3d2601758aa01b05bd4b7ae72dd17c100/packages/common/src/sort-by.ts#L64-L66